src/sgp/Smack_Stub.cc

```diff
@@ -37,38 +38,51 @@ BOOLEAN SmkAudioSwitch(smk SmkObj, BOOLEAN sw)
 }
 
 void SmackPrintFlickInfo(unsigned long width, unsigned long height, UCHAR8 scale, unsigned long framecount, 
-			DOUBLE usf, UCHAR8 a_channels, UCHAR8 a_depth, UCHAR8 a_rate)
+			DOUBLE usf, UCHAR8 a_channels, UCHAR8 a_depth, unsigned long a_rate)
 {
 	printf ("Video -- Frames: %lu Width: %lu Height: %lu Scale: %d \n", framecount, width, height, scale);
-	printf ("Audio -- FPS: %2.2f  Channels: %d Depth: %d Rate %d\n", usf/1000, a_channels, a_depth, a_rate);
+	printf ("Audio -- FPS: %2.2f  Channels: %u Depth: %u Rate %lu\n", usf/1000, a_channels, a_depth, a_rate);
 }
 
-// read all smackaudio and convert it to 22050Hz on the fly (44100 originally)
-UINT32 SmackGetAudio(const smk SmkObj, const INT16* audiobuffer)
+// read all smackaudio
+UINT32 SmackGetAudio(const smk SmkObj, UCHAR8** audiobuffer)
```

Comment:
Why does this use a pointer to a pointer?

Reply:
SmackGetAudio is the one creating the buffer now so it needs a way to return it along with the length.